src/antsibull_docs/schemas/docs/plugin.py

```diff
@@ -207,6 +210,16 @@ class PluginDocSchema(BaseModel):
 
 class PluginExamplesSchema(BaseModel):
     examples: str = ""
+    examples_format: str = "yaml"
+
+    @p.root_validator(pre=True)
+    # pylint:disable=no-self-argument
+    def extract_examples_format(cls, values):
+        if isinstance(values.get("examples"), str):
+            fmt_match = _EXAMPLES_FMT_RE.match(values["examples"].lstrip())
+            if fmt_match:
+                values["examples_format"] = fmt_match.group(1)
+        return values
```

Comment:
I would reweite this as a regular field validator and maybe use the walrus syntax for the regex match if that's your thing:

```suggestion
    @p.validator(pre=True, always=True)
    # pylint:disable=no-self-argument
    def extract_examples_format(cls, value: t.Any, values: dict[str, Any]):
        if isinstance(value, str) and fmt_match := _EXAMPLES_FMT_RE.match(values["examples"].lstrip()):
            value = fmt_match.group(1)
        return value
```

Reply:
I replaced it, though I had to modify your code a bit. `:=` does not work in `if X and Y := Z:`, that results in a syntax error.

Reply:
If you use `if isinstance(...) and (fmt_match := _EXAMPLES_FMT_RE.match(values["examples"].lstrip())` with parentheses around the second expression, it should work.

I'm not a huge fan of


``` python
if CONIDTION:
   if CONDITION2:
       ...
...
```

Other than that, this looks good to me.

Reply:
Due to the line length limit this gets reformatted as:
```py
        if isinstance(examples := values.get("examples"), str) and (
            fmt_match := _EXAMPLES_FMT_RE.match(examples.lstrip())
        ):
            value = fmt_match.group(1)
```
Which IMO is less readable than two nested `if`s.